index.js

```diff
@@ -1,53 +1,42 @@
 const dogs = []
-document.addEventListener('DOMContentLoaded', () => {
-    fetch('http://localhost:3000/dogs') 
+document.addEventListener("DOMContentLoaded", () => {
+    fetch("http://localhost:3000/dogs")
     .then(response => response.json())
     .then(data => getDogs(data))
    
-    const searchForm = document.getElementById("search-form");
+    const searchForm = document.getElementById("search-form")
     searchForm.addEventListener("submit", (e) => {searchDogs(e)})    
    
     const allDogsButton = document.getElementById("all")
-    allDogsButton.addEventListener("click", () => {showAllDogs()})
-    
+    allDogsButton.addEventListener("click", () => {
+        const dc = document.getElementById("dogs-container")
+        if (dc.children.length == 0) {
+            showAllDogs()
+        }
+    })
 })
 
-function getDogs(data) {
-    data.forEach((dog) => dogs.push(dog))
-    
+function getDogs(dogData) {
+    dogData.forEach((dog) => dogs.push(dog))
 }
 
 function searchDogs(e) {
-    e.preventDefault();
-    const findDogs = dogs.filter(dog => dog.dogBreed === e.target.dogBreed.value)
-    findDogs.forEach((dog) => {
-        const h2 = document.createElement("h2");
-        h2.innerText = dog.name;
-        const dogContainer = document.getElementById("searchByDogBreed_container")
-        const img = document.createElement("img")
-        img.src = dog.image
-        img.className = "dog-img"
-        dogContainer.append(h2, img)
-    })
+    e.preventDefault()
```

Comment:
This is the big change! It's also okay that you didn't do this.

I noticed that there was a lot of duplicate code in `searchDogs` and `showAllDogs`. They are essentially the same function, just with a different data set and a different place to append the data.

I moved the duplicate code to a new function, `createDogElement`. When you call your `forEach` in either function, it passes each dog to the `createDogElement` one at a time. 

You'll notice that it takes two inputs, too. The second one is the name of the container that the dog needs to be attached to.